crates/bevy_ui/src/node_bundles.rs

```diff
@@ -153,9 +202,39 @@ impl TextBundle {
         self
     }
 
-    /// Returns this [`TextBundle`] with a new [`Style`].
-    pub const fn with_style(mut self, style: Style) -> Self {
-        self.style = style;
+    /// Returns this [`TextBundle`] with a new [`FlexContainer`].
+    pub const fn with_layout(mut self, layout: FlexContainer) -> Self {
```

Comment:
I'm not sure I like this. `TextBundle` is slowly building it's own api, but it's the only bundle used in bevy with this `with_` api. I didn't mind it that much when it was just `with_style()` but it's really starting to look like a missing abstraction.

Reply:
I agree. Now that we have Components-as-bundles, we should remove this API altogether.

Reply:
Yeah, I think I want to remove these.

Reply:
PR has been updated, this has now been resolved.